README.md

```diff
@@ -33,29 +33,44 @@ Authorization code provider is used by Web Apps (ASP.NET & ASP.NET Core) to acqu
 It uses [MSALs Authorization Code](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Acquiring-tokens-with-authorization-codes-on-web-apps) to authenticate Microsoft Graph requests.
 
 ```csharp
-IConfidentialClientApplication clientApplication = AuthorizationCodeProvider.CreateClientApplication(clientId, redirectUri, clientCredential);
-
-AuthorizationCodeProvider authenticationProvider = new AuthorizationCodeProvider(clientApplication, scopes);
+IConfidentialClientApplication confidentialClientApplication = ConfidentialClientApplicationBuilder
+                .Create(clientId)
+                .WithRedirectUri(redirectUri)
+                .WithClientSecret(clientSecret)
+                .WithAuthority(authority)
+                .Build();
+
+AuthorizationCodeProvider authenticationProvider = new AuthorizationCodeProvider(confidentialClientApplication, scopes);
 ```
 
 #### b. Client credential provider
 Client credential provider is used by services and desktop applications to acquire Microsoft Graph access token without a user. The app should have previously registered a secret (app password or certificate) with Azure AD during the application registration.
 This provider leverages on [MSALs Client Credential Flows](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Client-credential-flows) to authenticate Microsoft Graph requests.
 
 ```csharp
-IConfidentialClientApplication clientApplication = ClientCredentialProvider.CreateClientApplication(clientId, clientCredential);
-
-ClientCredentialProvider authenticationProvider = new ClientCredentialProvider(clientApplication);
+IConfidentialClientApplication confidentialClientApplication = ConfidentialClientApplicationBuilder
+                .Create(clientId)
+                .WithRedirectUri(redirectUri)
```

Comment:
Is RedirectUri required here?

Reply:
With MSAL v3, you need to pass either a tenantId via `.WithTenantId("{tenantID}")` or a redirectUrl with a \tenantId; Otherwise, it will throw an exception with error code `Authorization_IdentityNotFound`. I'll update the readme to use `.WithTenantId("{tenantID}")` instead.

Reply:
The same holds true for IWA provider and UsernamePassword provider.